internal/runner/runnerpool/runner.go

```diff
@@ -175,14 +245,90 @@ func (r *Runner) Run(ctx context.Context, l log.Logger, opts *options.Terragrunt
 
 	r.queue.FailFast = opts.FailFast
 	r.queue.IgnoreDependencyOrder = opts.IgnoreDependencyOrder
+	// Allow continuing the queue when dependencies fail if requested via CLI
+	r.queue.IgnoreDependencyErrors = opts.IgnoreDependencyErrors
 	controller := NewController(
 		r.queue,
 		r.Stack.Units,
 		WithRunner(task),
 		WithMaxConcurrency(opts.Parallelism),
 	)
 
-	return controller.Run(ctx, l)
+	err := controller.Run(ctx, l)
+
+	// Emit report entries for early exit units after controller completes
+	if r.Stack.Report != nil {
+		for _, entry := range r.queue.Entries {
+			if entry.Status == queue.StatusEarlyExit {
+				unit := r.getUnitByPath(entry.Config.Path)
+				if unit == nil {
+					l.Warnf("Could not find unit for early exit entry: %s", entry.Config.Path)
+					continue
+				}
+
+				// Ensure path is absolute for reporting
+				unitPath := unit.Path
+				if !filepath.IsAbs(unitPath) {
+					var absErr error
+
+					unitPath, absErr = filepath.Abs(unitPath)
+					if absErr != nil {
+						l.Errorf("Error getting absolute path for unit %s: %v", unit.Path, absErr)
+						continue
+					}
+				}
+
+				run, reportErr := r.Stack.Report.EnsureRun(unitPath)
+				if reportErr != nil {
+					l.Errorf("Error ensuring run for early exit unit %s: %v", unitPath, reportErr)
+					continue
+				}
+
+				// Find the immediate failed or early-exited ancestor to set as cause
+				// If a dependency failed, use it; otherwise if a dependency exited early, use it
+				var (
+					failedAncestor string
+					foundFailedDep bool
+				)
+
+				for _, dep := range entry.Config.Dependencies {
+					// Search for the dependency in queue entries
+					for _, qEntry := range r.queue.Entries {
+						if qEntry.Config.Path == dep.Path {
+							if qEntry.Status == queue.StatusFailed {
+								failedAncestor = filepath.Base(dep.Path)
+								foundFailedDep = true
+
+								break
+							} else if qEntry.Status == queue.StatusEarlyExit && failedAncestor == "" {
```

Comment:
This doesn't have to be an else if, it can be an if. 

If we take the previous branch, we break.